akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/EventSourcedBehavior.scala

```diff
@@ -256,4 +261,14 @@ object EventSourcedBehavior {
    * If not defined, the default `akka.persistence.typed.stash-capacity` will be used.
    */
   def withStashCapacity(size: Int): EventSourcedBehavior[Command, Event, State]
+
+  /**
+   * Invoke this callback when an event from another replica arrives, delaying persisting the event until the returned
+   * future completes, if the future fails the actor is crashed.
+   *
+   * Only used when the entity is replicated.
+   */
+  @ApiMayChange
+  def withReplicatedEventInterceptor(
+      interceptor: (ReplicaId, Long, State, Event) => Future[Done]): EventSourcedBehavior[Command, Event, State]
```

Comment:
would it be possible to use the `ReplicationContext` instead of some of these parameters, or use ReplicationContext as a parameter to the function?

Reply:
I started like that but it's stateful and has threading protection, so then we need some before-after cleanup and easy for user to anyway accidentally use it from within a Future which would throw.

Reply:
We can make it a SAM for Scala as well for the many-unnamed parameters problem but sure would be nicer if ReplicationContext was immutable and could be safely passed

Reply:
Accessing the pid means closing over it or the replication context (where it is a val) in Scala or calling this.persistenceId() from Java, but might be more convenient to have it available through a parameter instead.

Reply:
Closing over pid worked for our own internal use case at least.

Adding a final class for the parameter could make evolving the interceptor easier, in case we realize something is missing, but that also means allocating it for every incoming event when replicating and there is an interceptor. Not sure the risk that we will need to add some more parameter is big enough.